immuneML/reports/encoding_reports/PositionalMotifFrequencies.py

```diff
@@ -38,41 +41,63 @@ def _generate(self):
         PathBuilder.build(self.result_path)
 
         motifs = self.dataset.encoded_data.feature_names
-        plotting_data = self._get_plotting_data(motifs)
+        stacked_bar, hist_dict = self._get_plotting_data(motifs)
 
-        report_output_fig = self._safe_plot(stacked_bar=plotting_data)
-        output_figures = None if report_output_fig is None else [report_output_fig]
+        output_figures = self._safe_plot(stacked_bar=stacked_bar, hist_dict=hist_dict)
         return ReportResult(name=self.name, output_figures=output_figures)
 
+
     def _get_plotting_data(self, motifs):
         positional_aa_counts = {aa: [0 for i in range(self.get_sequence_length())]
                                 for aa in EnvironmentSettings.get_sequence_alphabet(SequenceType.AMINO_ACID)}
 
+        distance_aa_counts = {position: {distance: 0 for distance in range(1, self.get_sequence_length())} for position in range(self.get_sequence_length())}
+
         for motif in motifs:
             indices, amino_acids = PositionalMotifHelper.string_to_motif(motif, "&", "-")
+            distance = len(indices)
+
+            for i in range(distance-1):
+                distance_aa_counts[distance][indices[i+1]-indices[i]] += 1
 
             for index, amino_acid in zip(indices, amino_acids):
                 positional_aa_counts[amino_acid][index] += 1
 
+        hist_dict = dict()
+
+        for distance_count in distance_aa_counts:
+            if sum(distance_aa_counts[distance_count].values()) != 0:
+                hist_dict[distance_count] = distance_aa_counts[distance_count]
+
         stacked_bar = pd.DataFrame(positional_aa_counts)
         stacked_bar = stacked_bar.loc[:, (stacked_bar != 0).any(axis=0)]
 
-        return stacked_bar
+
+        return stacked_bar, hist_dict
 
     def _get_color_discrete_sequence(self):
         return px.colors.qualitative.Pastel[:-1] + px.colors.qualitative.Set3
 
-    def _plot(self, stacked_bar) -> ReportOutput:
+    def _plot(self, stacked_bar, hist_dict) -> ReportOutput:
```

Comment:
it would be nice if each plot has its own method. 
i.e., _plot(.., ...) calls two methods:
_plot_motif_frequences(...)
_plot_gap_sizes(...)

the responsibility of _plot is then only to collect the results of these other functions and return a list with ReportOutput objects. If no plots were created, it should return None (never an empty list) -> that's a bit immuneML-specific, but if an empty list is returned it will show up weird in the output HTML.